scripts/modules/models-cart.js

```diff
@@ -246,8 +246,49 @@ define(['underscore', 'modules/backbone-mozu', 'hyprlive', "modules/api", "modul
             });
         },
         toJSON: function(options) {
-            var j = Backbone.MozuModel.prototype.toJSON.apply(this, arguments);
+            var self = this;
+            var j = Backbone.MozuModel.prototype.toJSON.apply(this, arguments); 
+            if (j && j.discountThresholdMessages && self.bfCurrencyConverter(j)) {
+                j.discountThresholdMessages = self.bfCurrencyConverter(j);
+            }
             return j;
+        },
+        bfCurrencyConverter: function(model) {
+            try{
+                //Check if BF is enabled
+                var pageContext = require.mozuData('pagecontext');
+                var isBfEnabled = $.cookie('currency_country_code');
+                if(isBfEnabled && isBfEnabled !== "US"){
+                    //Update currency rate for BF.
+                    var currencyRate = pageContext.currencyRateInfo.rate;
+                    var currencySymbol = pageContext.currencyInfo.symbol;
+                    var discountThresholdMessages = model.discountThresholdMessages;
+                    //check if threshold message is available
+                    if(discountThresholdMessages){
+                        for(var p = 0; p < discountThresholdMessages.length; p++){
```

Comment:
You could simply this using the each method in underscore. https://underscorejs.org/#each We use this library heavily throughout the app.